src/app/OperationalSessionSetup.h

```diff
@@ -326,6 +327,8 @@ class DLL_EXPORT OperationalSessionSetup : public SessionDelegate,
      * This function will set new IP address, port and MRP retransmission intervals of the device.
      */
     void UpdateDeviceData(const Transport::PeerAddress & addr, const ReliableMessageProtocolConfig & config);
+
+    ObjectPool<AddressResolve::ResolveResult, Dnssd::CommonResolutionData::kMaxIPAddresses> mDiscoveredResults;
```

Comment:
I suspect the BSS problem is that ResolveResult is not that small, we have a pool of them here, and we have a pool of OperationalSessionSetup objects, so we end up with lots of these ResolveResults allocated....

How much RAM are we willing to spend on this?  Unlike commissionable discovery, operational discovery has to be done by constrained devices...

Reply:
I don't need to hold "ResolveResult". I have changed the code with a struct that is smaller:
```
struct DeviceData
{
    Transport::PeerAddress address;
    ReliableMessageProtocolConfig mrpRemoteConfig;

    DeviceData() : address(Transport::Type::kUdp), mrpRemoteConfig(GetDefaultMRPConfig()) {}
};

ObjectPool<DeviceData, Dnssd::CommonResolutionData::kMaxIPAddresses> mDiscoveredDeviceData;
```

That said, I still suspect the BSS increase to be fairly large. We won't hold a pool of struct that contains a pool, but the first pool is size `64` by default https://github.com/project-chip/connectedhomeip/blob/f54f0f6a46f849fecafc26bb44652a5c4f0f0fec/src/lib/core/CHIPConfig.h#L946 which is already a lot for constrained device.

Reply:
Ah I have read the wrong macro. It should be `4` by default for those device, which is much more reasonable.
```
https://github.com/project-chip/connectedhomeip/blob/f54f0f6a46f849fecafc26bb44652a5c4f0f0fec/src/lib/core/CHIPConfig.h#L983
```

Reply:
I have updated the code such that it works a bit differently. The reason behind this change is because I found that a `ResolveResult` struct was already defined in `mAddressLookupHandle` which was resulting into doubling the additional memory.
So, instead of making a local copy on `src/app/OperationalSessionSetup.*`, the address lookup handle keeps a list of results (configurable by a flag) and returns them as needed.